### DISCUSSION TITLE
스냅샷은 어떤 기준으로 생성하는 것이 베스트플랙티스 일까요?

### DISCUSSION THREAD

**Discussion by LenKIM** [naver/scavenger#100] [General]
스냅샷은 어떤 기준으로 생성하는 것이 베스트플랙티스 일까요?

**Comment by taeyeon-Kim** [comment]
안녕하세요 :)
스냅샷의 생성의 best practice는 제 경험상 `애플리케이션`과 `환경`을 적절하게 조합하는게 중요해보입니다.
적절하게라는 의미가 애매할 수 있는데요.
A라는 메서드를 데드 코드 여부를 판단하고자 가정해보겠습니다.
A 메서드는 core 모듈에 위치해 있고 API와 batch 모듈은 각각 core 모듈을 사용하고 있다고도 가정하겠습니다.
그러면 A 메서드의 호출 여부를 파악하기 위해서는 `애플리케이션`에 API, batch 모듈을 모두 체크해야지 A메서드가 호출되고 있는지의 판별을 정확하게 할 수 있습니다.
같은 이치로 `환경`또한 개발, 운영을 모두 체크해야지만  A메서드가 호출되고 있는지의 판별을 정확하게 할 수 있겠죠

이렇게.. 내가 어떤 목적에 의해 스냅샷을 생성할지를 정하면 `애플리케이션`과 `환경`뿐 아니라 `처리 기간`도 설정해서 메서드 호출 유무를 파악할 수 있을 거라 보입니다.

**Reply by LenKIM** [reply]
그렇군요. 예를 들어 Order Application 과  Seller Application 있고, 각 `Application`은 DEV, PROD 로 구분될 수 있습니다. 이때 각 `Application` 이 호출하고 있는 `모듈`이 A,B,C 각자 다르게 호출되고 있다고 가정합니다.

이 때 우리의 목표는 각 `모듈`에서 사용되어지지 않는 죽은 코드를 찾는 것을 목표로 합니다. 그렇게 하기 위해서는 스냅샵의 생성이름을 {application}-{env} 이런 식으로 조합해야 찾을 수 있다는 말씀이신거죠?

**Reply by taeyeon-Kim** [reply]
아뇨 그렇지않습니다. 아래 스크린샷처럼 한 스냅샷에 모든 application과 env를 조합할 수 도 있습니다.
<img width="558" alt="image" src="https://github.com/naver/scavenger/assets/27491403/e619dad1-c27b-4238-b276-b810121182ee">

**Reply by LenKIM** [reply]
아하~ 이해했습니다. 그렇다면 이름은 어떤 기준으로 하는 것을 선호하시나요?

**Reply by taeyeon-Kim** [reply]
사용하는 서비스의 특성마다 다릅니다만.. 
말씀주신 케이스에서는 각 모듈(애플리케이션)별로 이름을 생성해서 사용할 것 같습니다.
스캐빈저를 예로 들면 각 모듈 별로 모든 환경은 포함시켜서 collector, api 이렇게 생성하겠네요 :)
